pyblish/plugin.py

```diff
@@ -666,6 +671,15 @@ def __call__(self, key=None, default=None):
 
         return self.get(key, default)
 
+    def __setitem__(self, k, v):
+        # Backwards incompatible data validation.
+        if STRICT_DATATYPES:
```

Comment:
For perfect compatibility, you could put the if-statement *outside* of the method declaration. That way, the override would only happen for early adopters, avoiding potential edge-cases where overriding this causes trouble.